bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -530,6 +526,19 @@ pub mod pallet {
 				)?;
 			}
 
+			if let Some(version_execution_header) = &update.execution_header {
+				let finalized_block_root: H256 = update
+					.finalized_header
+					.hash_tree_root()
+					.map_err(|_| Error::<T>::HeaderHashTreeRootFailed)?;
+				Self::store_execution_header(
```

Comment:
I think it's still nessessary. As you can see we `fetchEthereumNonce` 
https://github.com/Snowfork/snowbridge/blob/e8557c79fa5e20142cb6ae053ce33a3f3a517623/relayer/relays/execution/main.go#L116 at the `LastExecutionHeaderState` which should be a state finalized.

Reply:
Ah I see, good point. 👍🏻 

What do you think about potentially importing the same execution header twice? Maybe we should check if it exists in `process_execution_header_update`?

Reply:
Potentially, and I add some extra check here
https://github.com/Snowfork/snowbridge/blob/e8557c79fa5e20142cb6ae053ce33a3f3a517623/relayer/relays/beacon/header/header.go#L353-L360 to make sure `SyncExecutionHeader` work as expected.

Reply:
That check isn't related to checking if the header has already been imported on the light client side though. I think in `process_execution_header_update` we should check if the execution header has already been imported, before we try and verify it. What do you think?

Reply:
> An attacker can import execution header number 0 into the light client and then effectively force other relayers to resume relaying execution headers after number 0

Nice catch, made a fix in https://github.com/Snowfork/polkadot-sdk/pull/123/commits/6d8536d03836fe325590adf8e156fe4f79173d38

Also good point about `remove LastExecutionHeaderState storage item completely` and  `the relayers to store a local file on disk containing the block number they need to resume relaying from`,  but I'd rather improve that in a separate PR later.

Reply:
Thanks Ron, though 6d8536d doesn't solve all the problems with LastExecutionHeaderState.

Also, circling back to the original comments in this thread:

> Clara: Do we need to verify and store these executions at all?

> Ron: I think it's still nessessary. As you can see we fetchEthereumNonce
https://github.com/Snowfork/snowbridge/blob/e8557c79fa5e20142cb6ae053ce33a3f3a517623/relayer/relays/execution/main.go#L116 at the LastExecutionHeaderState which should be a state finalized.

My understanding from this that we are including the execution header in the `Update` struct passed to `submit(update)`, purely so that `LastExecutionHeaderState` can be initialized? If that's the case, we can't do that because `LastExecutionHeaderState` needs to be removed in any case.

Or is there another reason why `submit_execution_header` cannot be the only way of submitting execution headers?